src/components/field-types/DateFieldComponent.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <validate :state="state" :custom="{'validate': isValidDate(localValue) && validate(field)}">
+    <div class="form-group">
+      <label :for="field.id">{{ field.label }}</label>
+
+      <div class="input-group">
+        <flat-pickr
+          :id="field.id"
+          v-model="localValue"
+          :config="config"
+          placeholder="Select date"
```

Comment:
I have held back on the placeholders because
1) current forms do not have them
2) i18n is an issue